### PR TITLE
Staging hfcl ion4 target addition

### PR DESCRIPTION
**Pull request by kaushalendra581**
Added patch for HFCL-ion4

### CODE REVIEW

**Review by cbuschfb — COMMENTED** [review]
@ArifAlam can you please have a review as this PR is for a 1.x OpenSync use, Thanks Chris

**Comment by cbuschfb** [comment]
@ArifAlam request pull review for 1.x OpenSync use

**Comment by ArifAlam** [comment]
Few things:
- Please add the new AP to the build by modifying these files: wlan-ap/.github/workflows/build-dev.yml  and build.yml
- Please just have one commit in the PR.

**Comment by kaushalendra581** [comment]
@ArifAlam Recommended changes are incorporated please review and merge.

**Comment by ArifAlam** [comment]
Ensure that the staging build for this AP shows up on artifactory, that you can download that image from artifactory, upgrade the AP with it and boot it up with no issues.

**Comment by siddharthpandey02** [comment]
@ArifAlam Recommended changes are incorporated please review and merge.
We have also verified FW image from following path as suggested.
https://tip.jfrog.io/ui/native/tip-wlan-ap-firmware/ex447/dev/hfcl_ion4-2021-09-04-staging-HFCL-ION4-Target-addition-a0a9922.tar.gz.

File: patches/0062-Subject-PATCH-HFCL-ION4-Target-addition-patch.patch [hunk 1]
```diff
@@ -0,0 +1,396 @@
...
+diff --git a/target/linux/ipq40xx/base-files/etc/hotplug.d/firmware/11-ath10k-caldata b/target/linux/ipq40xx/base-files/etc/hotplug.d/firmware/11-ath10k-caldata
+index 38f248f..57ddc8a 100644
+--- a/target/linux/ipq40xx/base-files/etc/hotplug.d/firmware/11-ath10k-caldata
++++ b/target/linux/ipq40xx/base-files/etc/hotplug.d/firmware/11-ath10k-caldata
+@@ -202,6 +202,9 @@ case "$FIRMWARE" in
+ 		ath10kcal_extract "0:ART" 4096 12064
+ 		ath10kcal_patch_mac_crc $(macaddr_add "$(cat /sys/class/net/eth0/address)" 9) #2.4G, wlan1
+ 		;;
++    hfcl,ion4)
++        ath10kcal_extract "0:ART" 4096 12064
```
Comment (blogic) [inline patches/0062-Subject-PATCH-HFCL-ION4-Target-addition-patch.patch:94]: bad indenting

File: patches/0062-Subject-PATCH-HFCL-ION4-Target-addition-patch.patch [hunk 2]
```diff
@@ -0,0 +1,396 @@
...
++        ;;
+ 	esac
+ 	;;
+ "ath10k/pre-cal-ahb-a800000.wifi.bin")
+@@ -268,6 +271,9 @@ case "$FIRMWARE" in
+ 		ath10kcal_extract "ART" 20480 12064
+ 		ath10kcal_patch_mac_crc $(macaddr_add $(cat /sys/class/net/eth0/address) -1)
+ 		;;
++    hfcl,ion4)
++        ath10kcal_extract "0:ART" 20480 12064
```
Comment (blogic) [inline patches/0062-Subject-PATCH-HFCL-ION4-Target-addition-patch.patch:104]: here aswell

File: profiles/hfcl_ion4.yml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+profile: hfcl_ion4
+target: ipq40xx
+subtarget: generic
+description: Build image for the HFCL ION4
+image: bin/targets/ipq40xx/generic/openwrt-ipq40xx-generic-hfcl_ion4-squashfs-nand-sysupgrade.bin
+packages:
+    - uboot-envtools
```
Comment (blogic) [inline profiles/hfcl_ion4.yml:7]: this should got into DEFAULT_PACKAGES in the image/Makefile

Comment (siddharthpandey02) [reply profiles/hfcl_ion4.yml:7]: @blogic sure we will move this to image/Makefile.
Please let us know if futher any changes required.

File: patches/0062-Subject-PATCH-HFCL-ION4-Target-addition-patch.patch (hunk 1 above)
Comment (siddharthpandey02) [reply patches/0062-Subject-PATCH-HFCL-ION4-Target-addition-patch.patch:94]: This is due tab space we will resolve this.

**Comment by siddharthpandey02** [comment]
@blogic Recommended changes are incorporated please review and merge.

**Comment by ArifAlam** [comment]
Merged to Rel1 pending.

**Pull request closed without merging**